lib/ansible/template/native_helpers.py

```diff
@@ -43,17 +76,31 @@ def ansible_eval_concat(nodes):
     if not head:
         return ''
 
+    unsafe = False
+
     if len(head) == 1:
         out = head[0]
+        unsafe = _is_unsafe(out)
 
         if isinstance(out, NativeJinjaText):
+            if unsafe:
```

Comment:
This check/wrap is redundant, since that wrapper is guaranteed to be a string- if `is_unsafe` is true, it's already correctly marked.

Reply:
Good point, I'll move the check for after this one and in the native concat code.